### ISSUE TITLE
Gene extractor doesn't label flora disks correctly

### ISSUE THREAD

**Issue by chaoticidealism** [Aurorastation/Aurora.3#15357]
When using the machine to extract plant genes in Xenoflora, the disk is supposed to be labeled with the plant variety and the gene, but it only lists the plant variety and a comma. The spot where the specific gene used to be is empty.

**Comment by chaoticidealism** [comment]
Still an issue! Screenshot taken recently shows that the description of the disk also has the gene missing:

<img width="283" height="137" alt="Image" src="https://github.com/user-attachments/assets/eba8ecac-973d-43ee-a7f2-4a6a7d62fb7a" />

**Comment by chaoticidealism** [comment]
Leaving a comment. Still a thing. I know it's minor, but it bugs me all to heck when I play xenobotany.

**Comment by Batrachophreno** [comment]
Recently refactored Xenobotany device code- confirmation this is still an issue?

**Comment by chaoticidealism** [comment]
> Recently refactored Xenobotany device code- confirmation this is still an issue?

Yes, still an issue. It's the same as it was before your changes.

**Comment by Batrachophreno** [comment]
This is insane plant genetics nightmare code. Notes to self:
SUBSYSTEM_DEF(plants)
- var/list/gene_tag_masks
- gene_tag_masks[gene_mask] = gene_tag

need to expand /client/proc/show_plant_genes()

**Comment by Batrachophreno** [comment]
Delay on this one but tracked it down. Fix PR shortly
<img width="752" height="63" alt="Image" src="https://github.com/user-attachments/assets/af4fc09a-65c8-4eb0-9bea-40e1c75a0646" />
